velox/exec/tests/SharedArbitratorTest.cpp

```diff
@@ -3628,131 +3865,68 @@ DEBUG_ONLY_TEST_F(SharedArbitrationTest, joinBuildSpillError) {
 }
 
 TEST_F(SharedArbitrationTest, concurrentArbitration) {
-  FLAGS_velox_suppress_memory_capacity_exceeding_error_message = true;
-  const int numVectors = 8;
-  std::vector<RowVectorPtr> vectors;
-  fuzzerOpts_.vectorSize = 32;
-  fuzzerOpts_.stringVariableLength = false;
-  fuzzerOpts_.stringLength = 32;
-  for (int i = 0; i < numVectors; ++i) {
-    vectors.push_back(newVector());
-  }
-  const int numDrivers = 4;
-  createDuckDbTable(vectors);
-
-  const auto queryPlan =
-      PlanBuilder()
-          .values(vectors, true)
-          .addNode([&](std::string id, core::PlanNodePtr input) {
-            return std::make_shared<FakeMemoryNode>(id, input);
-          })
-          .planNode();
-  const std::string referenceSQL = "SELECT * FROM tmp";
-
-  std::atomic<bool> stopped{false};
-
-  std::mutex mutex;
-  std::vector<std::shared_ptr<core::QueryCtx>> queries;
-  std::deque<std::shared_ptr<Task>> zombieTasks;
-
-  fakeOperatorFactory_->setAllocationCallback([&](Operator* op) {
-    if (folly::Random::oneIn(4)) {
-      auto task = op->testingOperatorCtx()->driverCtx()->task;
-      if (folly::Random::oneIn(3)) {
-        task->requestAbort();
-      } else {
-        task->requestYield();
-      }
+  for (const auto& capacities : std::vector<std::vector<uint64_t>>{
+           {16 * MB, 128 * MB},
+           {128 * MB, 16 * MB},
+           {128 * MB, 128 * MB},
+       }) {
+    auto totalCapacity = capacities[0];
+    auto queryCapacity = capacities[1];
+    setupMemory(totalCapacity);
+
+    FLAGS_velox_suppress_memory_capacity_exceeding_error_message = true;
+    const int numVectors = 8;
+    std::vector<RowVectorPtr> vectors;
```

Comment:
We can create the input once before the loop. Also not sure if you want to use newVectors() method in this test class? thanks!

Reply:
Fixed.